sync/syncer_test.go

```diff
@@ -249,10 +249,10 @@ loop:
 	}
 }
 
-func getPeersMock(peers []Peer) PeersImpl {
+func getPeersMock(peers []p2p.Peer) p2p.PeersImpl {
 	value := atomic.Value{}
 	value.Store(peers)
-	pm1 := PeersImpl{&value, nil}
+	pm1 := p2p.PeersImpl{Snapshot: &value, Exit: nil}
```

Comment:
encapsulate this if needed, there is no reason for snapshot and exit to be public

Reply:
Added `NewPeersImpl` constructor function. The fields are never read/written outside `PeersImpl` so initializing them via a constructor seems to be correct.